### DISCUSSION TITLE
Error Group Settings

### DISCUSSION THREAD

**Discussion by Gisselle-Guzman** [cisagov/ScubaGoggles#409] [Q&A]
Hi Community , 

I was wondering if anybody else has gotten this error message before . 

` sudo scubagoggles gws 
/usr/local/lib/python3.11/dist-packages/scubagoggles/provider.py:457: RuntimeWarning: Exception thrown while getting group settings; outputs will be incorrect: 'NoneType' object is not iterable
  warnings.warn(
`

![image](https://github.com/user-attachments/assets/c7e5ee1e-22cc-47f2-bdb9-861fdc01cb3c)

I tried looking at the the provider.py 

![image](https://github.com/user-attachments/assets/6eb585b5-d7e9-48dc-bed0-a43f4b1cc77f)

Seeing this is a group issue.  I do have the Group Settings API enabled and not seeing any issues there. 

![image](https://github.com/user-attachments/assets/99cac83c-19ca-437a-af8a-87ba168711eb)

From the provider.py script should I specify in the code the group to enumerate ? I think the tool is running successfully but before I present it to my boss want to make sure I am getting the details  `No relevant event in the current logs for the top-level OU, XXX.com. While we are unable to determine the state from the logs, the default setting is compliant; manual check recommended.  ` So far have had 3 test passed in Common Controls, Gmail, Rules in the baseline conformance reports. I received an 1 error for Groups in Business 

`
GWS.GROUPS.7.1v0.2	New Groups SHOULD be created with an Access type of Restricted unless necessary for agency mission fulfillment.	Error	Should	This test depends on the following API call(s) which did not execute successfully: [directory/v1/groups/list](https://developers.google.com/admin-sdk/directory/reference/rest/v1/groups/list), [groups-settings/v1/groups/get](https://developers.google.com/admin-sdk/groups-settings/v1/reference/groups/get). See terminal output for more details.` 

Any words of wisdom ?

**Comment by buidav** [comment]
Hello, 

Not sure of what's causing this issue on our end. Currently looking into it and trying to reproduce the same issue.
I've only seen this specific API error out when the Groups API was not enabled and someone tried running the tool. 

**Quick question for now, do you have at least one [Google Workspace group](https://support.google.com/a/answer/33329?hl=en) active in your GWS instance**?
